aptos-move/block-executor/src/executor.rs

```diff
@@ -1029,6 +1020,24 @@ where
             let res = executor.execute_transaction(&latest_view, txn, idx as TxnIndex);
             let must_skip = matches!(res, ExecutionStatus::SkipRest(_));
             match res {
+                ExecutionStatus::Abort(err) => {
+                    if let Some(commit_hook) = &self.transaction_commit_hook {
+                        commit_hook.on_execution_aborted(idx as TxnIndex);
+                    }
+                    alert!("Fatal VM error by transaction {}", idx as TxnIndex);
+                    // Record the status indicating the unrecoverable VM failure.
+                    return Err(SequentialBlockExecutionError::ErrorToReturn(BlockExecutionError::FatalVMError(err)));
+                },
+                ExecutionStatus::DelayedFieldsCodeInvariantError(msg) => {
+                    alert!("Discarding transaction because we got DelayedFieldsCodeInvariantError in the sequential fallback: {msg}");
+                    ret.push(E::Output::discard_output(StatusCode::DELAYED_MATERIALIZATION_CODE_INVARIANT_ERROR));
```

Comment:
We are changing the behavior here, and I'm not sure it's what we want to do? Resource group serialization is even more unlikely than these I think and it is also quite certainly constrained to a single transaction. If we see delayed field materialization issue, there is a bigger chance the whole block might have an issue, so returning with a block error seems safer to me (esp if we have a way to skip a block?)